lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3454,6 +3454,10 @@ namespace {
         ctors.push_back(createDefaultConstructor(Impl, result));
       }
 
+      // When a user creates a custom copy constructor, there is possibly (even
+      // probably) special semantics around copying members of the object. So,
+      // don't generate a value constructor which would allow the user to
+      // accidentally violate these.
```

Comment:
To be honest, this might just have been a rebasing error after landing the constructors patch. But I'll update it anyway (I like your version better).